service/topology_state_machine.cc

```diff
@@ -34,6 +35,49 @@ bool topology::contains(raft::server_id id) {
            left_nodes.contains(id);
 }
 
+bool topology::has_complete_feature_information() const noexcept {
+    return (normal_nodes.size() + transition_nodes.size() + new_nodes.size()) == supported_features.size();
```

Comment:
(Self-reviewing here)

This check is wrong. It is based on an assumption that all nodes in the cluster are already reflected in the `topology` table and, during the upgrade, the `supported_features` column of upgraded nodes will be gradually filled. However, this won't be true because, most likely, features in raft and topology changes on raft will be introduced in the same release.